cpp/src/io/text/multibyte_split.cu

```diff
@@ -266,8 +291,9 @@ __global__ void multibyte_split_kernel(
 
   // Step 5: Assign outputs from each thread using match offsets.
 
-  for (int32_t i = 0; i < ITEMS_PER_THREAD and i < thread_input_size; i++) {
-    if (trie.is_match(thread_states[i]) and not thread_offset.is_past_end()) {
+  for (int32_t i = 0; i < ITEMS_PER_THREAD; i++) {
+    auto const is_match = (thread_match_mask[i / 32] >> (i % 32)) & 1u;
```

Comment:
The `& 1u` seems unnecessary. Should we declare this as `bool const` instead of `auto`, or do a static cast?

Reply:
It's necessary to make sure we only read that single bit, otherwise we might also return `true` from more significant bits.

Reply:
Sorry, I misinterpreted this on my first glance. It's like `% 2`, not like `> 0`. Resolving.